src/databricks/labs/remorph/reconcile/connectors/oracle.py

```diff
@@ -1,56 +1,61 @@
-from pyspark.errors import PySparkException
-from pyspark.sql import DataFrame, DataFrameReader
+from pyspark.sql import DataFrame, DataFrameReader, SparkSession
+from sqlglot import Dialects
 
 from databricks.labs.remorph.reconcile.connectors.data_source import DataSource
+from databricks.labs.remorph.reconcile.connectors.jdbc_reader import JDBCReaderMixin
+from databricks.labs.remorph.reconcile.connectors.secrets import SecretsMixin
 from databricks.labs.remorph.reconcile.constants import SourceDriver, SourceType
 from databricks.labs.remorph.reconcile.recon_config import JdbcReaderOptions, Schema
+from databricks.sdk import WorkspaceClient
 
 
-class OracleDataSource(DataSource):
+class OracleDataSource(DataSource, SecretsMixin, JDBCReaderMixin):
+
+    def __init__(
+        self,
+        engine: Dialects,
+        spark: SparkSession,
+        ws: WorkspaceClient,
+        secret_scope: str,
+    ):
+        self._engine = engine
+        self._spark = spark
+        self._ws = ws
+        self._secret_scope = secret_scope
+
     @property
     def get_jdbc_url(self) -> str:
         return (
-            f"jdbc:{SourceType.ORACLE.value}:thin:{self._get_secrets('user')}"
-            f"/{self._get_secrets('password')}@//{self._get_secrets('host')}"
-            f":{self._get_secrets('port')}/{self._get_secrets('database')}"
+            f"jdbc:{SourceType.ORACLE.value}:thin:{self._get_secret_if_exists('user')}"
```

Comment:
Let's assume the Secrets doesn't exist: 
* We are still creating the `jdbc_url` like `jdbc:None:thin:None:____` 
* calling the `jdbc_reader` and reader load.
* In the `reader` load we are catching the `RunTimeError`. 

These steps can be totally avoided if we check for the `None` and stop further execution.